blockchain/blockchain.go

```diff
@@ -302,32 +320,56 @@ func (bc *BlockChain) migrateState(rootHash common.Hash) error {
 	var queue []common.Hash
 	committedCnt := 0
 
+	quitCh := make(chan struct{})
+	defer close(quitCh)
+
+	// TODO-Klaytn need to tune the threads / batch size
+	//threads := int(math.Max(math.Ceil(float64(runtime.NumCPU())/2), 4))
+	threads := int(math.Max(float64(runtime.NumCPU())-2, 4))
+
+	hashCh := make(chan common.Hash, threads*100)
+	resultCh := make(chan statedb.SyncResult, threads*100)
+
+	for th := 0; th < threads; th++ {
+		go bc.concurrentRead(srcCachedDB, quitCh, hashCh, resultCh)
+	}
+
 	for trieSync.Pending() > 0 {
 		bc.committedCnt, bc.pendingCnt = committedCnt, trieSync.Pending()
-		queue = append(queue[:0], trieSync.Missing(database.IdealBatchSize)...)
+		queue = append(queue[:0], trieSync.Missing(50000)...)
```

Comment:
What do you think about changing this number to `database.IdealBatchSize / threads`?

Reply:
I reverted this code.
Original size shows better performance.